apps/site/assets/ts/schedule/ScheduleLoader.tsx

```diff
@@ -0,0 +1,241 @@
+import React, { ReactElement } from "react";
+import { connect } from "react-redux";
+import { useQueryParams, StringParam } from "use-query-params";
+import SchedulePage from "./components/SchedulePage";
+import ScheduleNote from "./components/ScheduleNote";
+import ScheduleDirection from "./components/ScheduleDirection";
+import { SchedulePageData, SelectedOrigin } from "./components/__schedule";
+import { MapData, StaticMapData } from "../leaflet/components/__mapdata";
+import ScheduleFinder from "./components/ScheduleFinder";
+import ScheduleFinderModal from "./components/schedule-finder/ScheduleFinderModal";
+import { DirectionId } from "../__v3api";
+import mapStateToProps, {
+  getCurrentState,
+  storeHandler
+} from "./ScheduleStore";
+
+interface Props {
+  schedulePageData: SchedulePageData;
+  component: string;
+  updateURL: Function;
+  selectedDirection: DirectionId;
+}
+
+export const ScheduleLoader = ({
+  component,
+  schedulePageData,
+  updateURL,
+  selectedDirection
+}: Props): ReactElement<HTMLElement> => {
+  const [query] = useQueryParams({
+    // eslint-disable-next-line @typescript-eslint/camelcase
+    "schedule_finder[direction_id]": StringParam,
+    "schedule_finder[origin]": StringParam
+  });
+
+  const changeDirection = (direction: DirectionId): void => {
+    const currentState = getCurrentState();
+    const { modalOpen } = currentState;
+
+    storeHandler({
+      type: "CHANGE_DIRECTION",
+      newStoreValues: {
+        selectedDirection: direction,
+        selectedOrigin: null
+      }
+    });
+    if (!modalOpen) updateURL(direction, null);
+  };
+
+  const changeOrigin = (origin: SelectedOrigin): void => {
+    storeHandler({
+      type: "CHANGE_ORIGIN",
+      newStoreValues: {
+        selectedOrigin: origin
+      }
+    });
+    const currentState = getCurrentState();
+    const { selectedDirection: currentDirection } = currentState;
+    updateURL(currentDirection, origin);
+
+    // reopen modal depending on choice:
+    storeHandler({
+      type: "OPEN_MODAL",
+      newStoreValues: {
+        modalMode: origin === null ? "origin" : "schedule"
+      }
+    });
+  };
+
+  const closeModal = (): void => {
+    storeHandler({
+      type: "CLOSE_MODAL"
+    });
+    // clear parameters from URL when closing the modal:
+    updateURL("", "");
+  };
+
+  React.useEffect(() => {
+    // set initial state for the store when the component mounts
+    let newDirection = selectedDirection;
+    let newOrigin;
+    let modalMode = "origin";
+    let modalOpen = false;
+
+    if (query["schedule_finder[direction_id]"] !== undefined) {
+      newDirection = query["schedule_finder[direction_id]"] === "0" ? 0 : 1;
+    }
+    if (query["schedule_finder[origin]"] !== undefined) {
+      newOrigin = query["schedule_finder[origin]"];
+    }
+
+    const button = document.getElementById("modal-close")!;
```

Comment:
@alicia-mbta so basically here you're looking for the modal-close button as a way of determining if the modal is open?